src/mount.ts

```diff
@@ -482,7 +482,10 @@ export function mount(
         // otherwise we run into a proxy set error
         // due to https://github.com/vuejs/core/commit/f73925d76a76ee259749b8b48cb68895f539a00f#diff-ea4d1ddabb7e22e17e80ada458eef70679af4005df2a1a6b73418fec897603ceR404
         // introduced in Vue v3.2.45
-        if (hasSetupState(this)) {
+        // Also ensures not to include option API components in this this block
+        // since they can also have setup state but need to be patched using
+        // the regular method.
+        if (hasSetupState(this) && this.$.setupState.__isScriptSetup) {
```

Comment:
nit: typo `this this`

Reply:
Yup, you're right! I've removed the `hasSetupState` check from here because it's redundant, but we can't replace `hasSetupState` with `isScriptSetup` as they check for different things and some specs around expose fail if we dit.

Thanks for the review!